ruan_lao_shi_xiao_fen_dui/p2p-with-tracker/src/server/logic/node.rs

```diff
@@ -0,0 +1,61 @@
+use tracing::debug;
+
+use crate::server::error::*;
+use crate::server::store::{
+    piece, GLOBAL_TIMESTAMP_CACHE, NODE_INFO_CACHE, PIECE_NODE_INFO_CACHE, SESSION_INFO_CACHE,
+};
+use crate::server::utils::time_helper::get_current_timestamp;
+
+pub async fn update_session(node_id: &str) -> Result<()> {
+    let timestamp = get_current_timestamp();
+
+    // Check if node is the first to connect
+    #[cfg(debug_assertions)]
+    {
+        let is_empty = NODE_INFO_CACHE.is_empty();
+        if is_empty {
+            debug!(
+                "Node {} is first time to connect - timestamp:{}",
+                node_id, timestamp
+            );
+
+            // Update global time
+            GLOBAL_TIMESTAMP_CACHE.insert(timestamp);
```

Comment:
Could you tell me why did you insert the timestamp into it? By the way, what does `GLOBAL_TIMESTAMP_CACHE` do exactly?

Reply:
@Reto911 Hello, this cache is used in calc_download_time in `ruan_lao_shi_xiao_fen_dui/p2p-with-tracker/src/server/logic/file.rs` to test how long it takes the current node to download a file, it is done with a simple It is a simple list to access the timestamp information, and to calculate the time it takes to download a file, you just need to pop up the latest time and calculate the difference directly.

For multi-node testing, I didn't use this form directly, but instead calculated the time spent on calls to the synchronized HTTP interface.